.github/workflows/docs.yml

```
@@ -0,0 +1,693 @@
+          
+          // Migration guide
+          markdown += `## Migration Guide\n\n`;
+          markdown += `### v0.1.0 → v1.0.0 Changes\n\n`;
+          markdown += `This section will be updated as the migration progresses:\n\n`;
+          markdown += `- **Immutability**: All APIs now return immutable objects\n`;
+          markdown += `- **TypeScript**: Full TypeScript support with type definitions\n`;
+          markdown += `- **Modular Architecture**: Service-based architecture\n`;
+          markdown += `- **Error Handling**: Comprehensive error types and handling\n`;
+          markdown += `- **Validation**: Input validation with Zod schemas\n\n`;
+          
+          markdown += `### Breaking Changes\n\n`;
+          markdown += `*Breaking changes will be documented here as they are identified.*\n\n`;
+          
+          markdown += `### Compatibility Layer\n\n`;
+          markdown += `A compatibility layer will be provided to ease migration from v0.1.0 to v1.0.0.\n\n`;
+          
+          return markdown;
+        }
+        
+        // Generate documentation
+        const apiDocs = extractAPIDocumentation();
+        const markdown = generateMarkdownDocs(apiDocs);
+        
+        // Ensure docs directory exists
+        if (!fs.existsSync('docs')) {
+          fs.mkdirSync('docs', { recursive: true });
+        }
+        
+        // Write API documentation
+        fs.writeFileSync('docs/api-reference.md', markdown);
+        fs.writeFileSync('api-docs.json', JSON.stringify(apiDocs, null, 2));
+        
+        console.log('API documentation generated successfully');
+        console.log(`- Classes found: ${apiDocs.classes.length}`);
+        console.log(`- Methods found: ${apiDocs.methods.length}`);
+        EOF
+        
+        node extract-api-docs.js
+
+    - name: Generate TypeScript Documentation
+      run: |
+        echo "📘 Generating TypeScript documentation..."
+        
+        # Check if TypeScript files exist
+        if find src -name "*.ts" 2>/dev/null | grep -q .; then
+          echo "TypeScript files found, generating TypeDoc documentation..."
+          
+          # Generate TypeScript configuration if it doesn't exist
+          if [ ! -f "tsconfig.json" ]; then
+            echo "Creating tsconfig.json for documentation..."
+            cat > tsconfig.json << 'EOF'
+        {
+          "compilerOptions": {
+            "target": "ES2022",
+            "module": "ESNext",
+            "moduleResolution": "node",
+            "lib": ["ES2022"],
+            "outDir": "./dist",
+            "rootDir": "./src",
+            "strict": true,
+            "esModuleInterop": true,
+            "skipLibCheck": true,
+            "forceConsistentCasingInFileNames": true
+          },
+          "include": [
+            "src/**/*"
+          ],
+          "exclude": [
+            "node_modules",
+            "dist",
+            "**/*.test.*"
+          ]
+        }
+        EOF
+          fi
```

Don't overwrite the project's tsconfig.json. Use a dedicated documentation config:

```suggestion
            echo "Creating tsconfig.json for documentation..."
            cat > tsconfig.docs.json << 'EOF'
```

Then update the TypeDoc command to use this config:
```bash
npx typedoc --tsconfig tsconfig.docs.json
```